traitsui/qt4/tree_editor.py

```diff
@@ -272,6 +272,21 @@ def expand_levels(self, nid, levels, expand=True):
                 for cnid in self._nodes_for(nid):
                     self.expand_levels(cnid, levels - 1)
 
+    def expand_all(self):
+        """ Expands all expandable nodes in the tree.
+
+        Warning: If the Tree contains a large number of items, this function
+        will be very slow.
+        """
+        iterator = QtGui.QTreeWidgetItemIterator(self._tree)
+        while iterator.value():
+            item = iterator.value()
```

Comment:
I'm not sure how this iterator works but these lines threw me off completely. Code like this is usually broken because for normal python iterators we'll be iterating twice in a single loop, for each call to `iterator.value` but having tested the code in #673 using the fix you mentioned, I guess this works for this Qt iterator.

Please add any further links/documentation/comments you have regarding this iterator here - that'll hopefully help with understanding later on.

Reply:
It is a weird qt thing, that does things its own way.  The iteration is actually performed by the `iterator += 1` line below.  See docs https://doc.qt.io/qt-5/qtreewidgetitemiterator.html#operator-2b-eq

Reply:
I will add a link to https://doc.qt.io/qt-5/qtreewidgetitemiterator.html in a comment